portfolio/src/main/java/com/google/sps/data/MergeBooks.java

```diff
@@ -0,0 +1,67 @@
+package com.google.sps.data;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.data.Book;
+import java.util.ArrayList;
+
+public class MergeBooks {
+  /**
+   * This method merges two books that have the same title
+   * and ISBN and returns a new book combined from both of them
+   * @param googleApiBook: a book created by the bookResponse parser from the Books API
+   * Contains: title, categories, author, language, description, infoLink, pageCount, publishedDate,
+   * publisher, maturityRating, thumbnail, ISBN
+   * @param goodReadsBook: the book constructed from the csv file
+   * Contains: title, genre, reviews
+   */
+  public static Book merge(Book googleApiBook, Book goodReadsBook) {
+    validate(googleApiBook);
+    validate(goodReadsBook);
+    validateISBN(googleApiBook.isbn());
+    Preconditions.checkArgument(
+        googleApiBook.title().equals(goodReadsBook.title()), "The books must be of the same title");
+
+    Book.Builder combinedBuilder = Book.builder();
+    // Get API data
+    combinedBuilder.title(googleApiBook.title());
+    combinedBuilder.categories(new ArrayList<>(googleApiBook.categories()));
+    combinedBuilder.authors(new ArrayList<>(googleApiBook.authors()));
+    combinedBuilder.language(googleApiBook.language());
+    combinedBuilder.description(googleApiBook.description());
+    combinedBuilder.infoLink(googleApiBook.infoLink());
+    combinedBuilder.pageCount(googleApiBook.pageCount());
+    combinedBuilder.publishedDate(googleApiBook.publishedDate());
+    combinedBuilder.publisher(googleApiBook.publisher());
+    combinedBuilder.maturityRating(googleApiBook.maturityRating());
+    combinedBuilder.thumbnail(googleApiBook.thumbnail());
+    combinedBuilder.isbn(googleApiBook.isbn());
```

Comment:
You could replace this with:

`Book.Builder combinedBuilder = googleApiBook.toBuilder();`

Reply:
Great!